// <copyright file="MediaExtensionsTest.CreateRectangle.g.cs" company="SuprDewd">Copyright © SuprDewd 2010</copyright>
// <auto-generated>
// This file contains automatically generated unit tests.
// Do NOT modify this file manually.
// 
// When Pex is invoked again,
// it might remove or update any previously generated unit tests.
// 
// If the contents of this file becomes outdated, e.g. if it does not
// compile anymore, you may delete this file and invoke Pex again.
// </auto-generated>
using System;
using System.Drawing;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Pex.Framework.Generated;
using Microsoft.Pex.Engine.Exceptions;

namespace SharpBag.Media
{
    public partial class MediaExtensionsTest
    {
[TestMethod]
[PexGeneratedBy(typeof(MediaExtensionsTest))]
public void CreateRectangle621()
{
    Rectangle rectangle;
    rectangle = this.CreateRectangle(0, 0, 1, 1);
    Assert.AreEqual<int>(0, rectangle.X);
    Assert.AreEqual<int>(0, rectangle.Y);
    Assert.AreEqual<int>(1, rectangle.Width);
    Assert.AreEqual<int>(1, rectangle.Height);
    Assert.AreEqual<bool>(false, rectangle.IsEmpty);
}
[TestMethod]
[PexGeneratedBy(typeof(MediaExtensionsTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void CreateRectangleThrowsContractException295()
{
    try
    {
      Rectangle rectangle;
      rectangle = this.CreateRectangle(0, 0, 1, 0);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(MediaExtensionsTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void CreateRectangleThrowsContractException799()
{
    try
    {
      Rectangle rectangle;
      rectangle = this.CreateRectangle(0, 0, 0, 0);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
    }
}
